crossword.pl

```diff
@@ -46,10 +46,57 @@
 
 
 % Step 3
-%
 % Now, create crossword/10, a rule that will solve the crossword puzzle from the readme.
 %
-
+sixLetterWord(X) :- word(X,_,_,_,_,_,_).
+fiveLetterWord(X) :- word(X,_,_,_,_,_).
+sevenLetterWord(X) :- word(X,_,_,_,_,_,_,_).
+tenLetterWord(X) :- word(X,_,_,_,_,_,_,_,_,_,_).
+nineLetterWord(X) :- word(X,_,_,_,_,_,_,_,_,_).
+
+endsWith1(X,Y) :-
+	word(X,_,_,_,A),
+	word(Y,A,_,_,_,_,_).
+
+middleOf2(X,Y) :-
+	word(X,_,_,_,A,_,_),
+	word(Y,_,_,_,_,_,A).
+
+middleOf8(X,Y) :-
+	word(X,_,_,A,_,_),
+	word(Y,_,_,_,_,_,A).
+
+headOf3(X,Y) :-
+	word(X,_,A,_,_,_,_,_),
+	word(Y,_,A,_,_,_).
+
+tailOf3(X,Y) :-
+	word(X,_,_,_,_,_,A,_),
+	word(Y,_,_,_,A,_,_,_,_,_,_).
+
+headOf4(X,Y) :-
+	word(X,_,A,_,_,_,_,_),
+	word(Y,_,_,_,_,A).
+
+tailOf4(X,Y) :-
+	word(X,_,_,_,_,_,A,_),
+	word(Y,_,_,_,_,_,_,A,_,_,_).
+
+startOf9(X,Y) :-
+	word(X,A,_,_,_),
+	word(Y,A,_,_,_,_,_,_,_,_,_).
+
+crossword(Across1,Across6,Across7,Across8,Across9,Down2,Down3,Down4,Down5,Down9) :-
+	fourLetterWord(Across1), endsWith1(Across1,Down2),
+	sixLetterWord(Down2), middleOf2(Down2,Across7),
+	sevenLetterWord(Down3), headOf3(Down3,Across6), tailOf3(Down3,Across9),
+	sevenLetterWord(Down4), headOf4(Down4,Across6), tailOf4(Down4,Across9),
+	nineLetterWord(Down5),
+	fiveLetterWord(Across6),
+	sixLetterWord(Across7),
+	fiveLetterWord(Across8), middleOf8(Across8,Down2),
+	tenLetterWord(Across9),
+	fourLetterWord(Down9), startOf9(Down9,Across9).
 
```

Comment:
While it is possible to solve the entire crossword in a single rule, as in:

```prolog
crossword(Across1, Across6, Across7, Across8, Across9, Down2, Down3, Down4, Down5, Down9) :-
  word(Across1, _, _, _, A),
  word(Across6, _, B, _, _, C),
  word(Across7, _, D, _, _, _, E),
  word(Across8, _, _, F, _, _),
  word(Across9, G, _, _, H, _, _, I, _, _, J),
  word(Down2, A, _, _, E, _, F),
  word(Down3, _, B, _, _, _, H, _),
  word(Down4, _, C, _, _, _, I, _),
  word(Down5, _, _, D, _, _, J, _, _, _),
  word(Down9, G, _, _, _).
```

I like the fact that you broke the problem down into simpler problems.  This is a good approach to programming in general.